docs/diagrams/kubernetes_workflow.md

```diff
@@ -0,0 +1,1155 @@
+                KubeUnit->>KubeUnit: KubeLoggingWithReconnect()
+                Note over KubeUnit: See Diagram 5 for details
+            else No reconnect (K8s < 1.23.14)
+                KubeUnit->>KubeUnit: kubeLoggingNoReconnect()
+                KubeUnit->>KubeAPI: GetLogs(Follow=true,<br/>Timestamps=false)
+                KubeAPI->>LogStream: Open log stream
+                LogStream-->>KubeUnit: Stream handle
+                
+                KubeUnit->>LogStream: io.Copy(stdout, stream)
+                loop Read log stream
+                    LogStream->>KubeUnit: Log line
+                    KubeUnit->>StdoutFile: Write line
+                end
+                
+                alt Stream error
+                    LogStream-->>KubeUnit: Error (EOF or other)
+                    Note over KubeUnit: Log stream terminated<br/>(may be due to rotation or 4hr timeout)
+                end
+            end
+        end
+    else skipStdin (resume)
+        KubeUnit->>KubeUnit: UpdateBasicStatus(WorkStateRunning)
+        KubeUnit->>KubeAPI: GetLogs(Follow=true,<br/>Timestamps=true,<br/>SinceTime=lastTimestamp)
+        
+        loop Stream logs from sinceTime
+            KubeUnit->>KubeAPI: Read log lines
+            KubeAPI->>LogStream: Stream lines with timestamps
+            LogStream->>KubeUnit: Log line
+            KubeUnit->>KubeUnit: ProcessLogLine()<br/>(skip duplicates)
+            KubeUnit->>StdoutFile: Write new lines only
+        end
+    end
+
+    KubeUnit->>KubeUnit: Wait for both streams to complete
+    
+    alt Both streams successful
+        KubeUnit->>KubeUnit: UpdateFullStatus(WorkStateSucceeded)
+    else Error occurred
+        KubeUnit->>KubeUnit: UpdateBasicStatus(WorkStateFailed)
+    end
+```
+
+### Diagram 5: Logger Reconnection Logic
+
+This diagram details the sophisticated reconnection logic used in `KubeLoggingWithReconnect()` to handle log stream disconnections.
+
+```mermaid
+%%{init: {'theme':'base', 'themeVariables': { 'primaryColor':'#4a9eff', 'primaryTextColor':'#ffffff', 'primaryBorderColor':'#4a9eff', 'lineColor':'#6bb3ff', 'secondaryColor':'#1a1a1a', 'tertiaryColor':'#2a2a2a', 'background':'#000000', 'mainBkgColor':'#000000', 'secondBkgColor':'#1a1a1a', 'nodeBkg':'#1a1a1a', 'nodeBorder':'#4a9eff', 'clusterBkg':'#0a0a0a', 'clusterBorder':'#4a9eff', 'defaultLinkColor':'#6bb3ff', 'titleColor':'#ffffff', 'edgeLabelBackground':'#1a1a1a', 'actorBkg':'#000000', 'actorBorder':'#4a9eff', 'actorTextColor':'#ffffff', 'actorLineColor':'#4a9eff', 'signalColor':'#ffffff', 'signalTextColor':'#ffffff', 'labelBoxBkgColor':'#000000', 'labelBoxBorderColor':'#4a9eff', 'labelTextColor':'#ffffff', 'loopTextColor':'#ffffff', 'noteBkgColor':'#000000', 'noteTextColor':'#ffffff', 'noteBorderColor':'#ff6b35', 'activationBkgColor':'#1a1a1a', 'activationBorderColor':'#4a9eff', 'sequenceNumberColor':'#000000'}}}%%
+flowchart TD
+    Start([KubeLoggingWithReconnect]) --> MainLoop[Main reconnection loop]
```

Comment:
Is it this loop? https://github.com/ansible/receptor/blob/8b24174d290b3224ab51845830ed15728dd73635/pkg/workceptor/kubernetes.go#L376